policy.go

```diff
@@ -120,6 +120,11 @@ type BasePolicy struct {
 	// The default is false.
 	LinearizeRead bool
 
+	// When max connections are occupied in pool
```

Comment:
`WaitForConnectionsToBeRealeased` could be better named `WaitForConnectionsFromPool'

And the comments could be more clear + be a valid Go field comment

```
// WaitForConnectionsFromPool determines if a command that tries to get a 
// connection from the connection pool will wait and retry in case the pool is 
// empty until a connection becomes available (or the TotalTimeout is reached).  
// If set to true an error will be return immediately. 
// If set to false getting a connection will be retried.  
// This only applies if LimitConnectionsToQueueSize is set to true and the number of open connections to a node has reached ConnectionQueueSize.
// The default is true
```